src/lightning_lite/utilities/imports.py

```diff
@@ -0,0 +1,164 @@
+    False
+    """
+    try:
+        pkg = importlib.import_module(package)
+    except (ImportError, DistributionNotFound):
+        return False
+    try:
+        if hasattr(pkg, "__version__"):
+            pkg_version = Version(pkg.__version__)
+        else:
+            # try pkg_resources to infer version
+            pkg_version = Version(pkg_resources.get_distribution(package).version)
+    except TypeError:
+        # this is mocked by Sphinx, so it should return True to generate all summaries
+        return True
+    if use_base_version:
+        pkg_version = Version(pkg_version.base_version)
+    return op(pkg_version, Version(version))
+
+
+class _RequirementAvailable:
+    """Boolean-like class for check of requirement with extras and version specifiers.
+
+    >>> _RequirementAvailable("torch>=0.1")
+    Requirement 'torch>=0.1' met
+    >>> bool(_RequirementAvailable("torch>=0.1"))
+    True
+    >>> bool(_RequirementAvailable("torch>100.0"))
+    False
+    """
+
+    def __init__(self, requirement: str) -> None:
+        self.requirement = requirement
+
+    def _check_requirement(self) -> None:
+        if not hasattr(self, "available"):
+            try:
+                pkg_resources.require(self.requirement)
+                self.available = True
+                self.message = f"Requirement {self.requirement!r} met"
+            except Exception as ex:
+                self.available = False
+                self.message = f"Requirement {self.requirement!r} not met, {ex.__class__.__name__}: {ex}"
+
+    def __bool__(self) -> bool:
+        self._check_requirement()
+        return self.available
+
+    def __str__(self) -> str:
+        self._check_requirement()
+        return self.message
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+
+_IS_WINDOWS = platform.system() == "Windows"
+_IS_INTERACTIVE = hasattr(sys, "ps1")  # https://stackoverflow.com/a/64523765
+_PYTHON_GREATER_EQUAL_3_8_0 = (sys.version_info.major, sys.version_info.minor) >= (3, 8)
+_PYTHON_GREATER_EQUAL_3_10_0 = (sys.version_info.major, sys.version_info.minor) >= (3, 10)
+_TORCH_GREATER_EQUAL_1_9_1 = _compare_version("torch", operator.ge, "1.9.1")
+_TORCH_GREATER_EQUAL_1_10 = _compare_version("torch", operator.ge, "1.10.0")
+_TORCH_LESSER_EQUAL_1_10_2 = _compare_version("torch", operator.le, "1.10.2")
+_TORCH_GREATER_EQUAL_1_11 = _compare_version("torch", operator.ge, "1.11.0")
+_TORCH_GREATER_EQUAL_1_12 = _compare_version("torch", operator.ge, "1.12.0")
+_TORCH_GREATER_EQUAL_1_13 = _compare_version("torch", operator.ge, "1.13.0", use_base_version=True)
+
+_APEX_AVAILABLE = _module_available("apex.amp")
+_FAIRSCALE_AVAILABLE = not _IS_WINDOWS and _module_available("fairscale.nn")
+_FAIRSCALE_OSS_FP16_BROADCAST_AVAILABLE = _FAIRSCALE_AVAILABLE and _compare_version("fairscale", operator.ge, "0.3.3")
+_FAIRSCALE_FULLY_SHARDED_AVAILABLE = _FAIRSCALE_AVAILABLE and _compare_version("fairscale", operator.ge, "0.3.4")
```

Comment:
This PR is not updated to master changes, no?

#14476

Reply:
Updated in [731a59d](https://github.com/Lightning-AI/lightning/pull/14492/commits/731a59dfd7e8114660a2d27c71e9499460a1c679)

and also following commit